iota/commands/extended/get_inputs.py

```diff
@@ -38,13 +38,14 @@ def _execute(self, request):
     seed      = request['seed'] # type: Seed
     start     = request['start'] # type: int
     threshold = request['threshold'] # type: Optional[int]
+    security_level = request['securityLevel'] # type: Optional[int]
 
     # Determine the addresses we will be scanning.
     if stop is None:
       addresses =\
         [addy for addy, _ in iter_used_addresses(self.adapter, seed, start)]
     else:
-      addresses = AddressGenerator(seed).get_addresses(start, stop)
+      addresses = AddressGenerator(seed, security_level).get_addresses(start, stop)
```

Comment:
stuck with this one.  `test.commands.extended.get_inputs_test:GetInputsCommandTestCase.test_start` patches method with mock function, while in case of test for default use of `security_level`  it doesn't work like that. Whole `AddressGenerator` class must be patched (or at least I don't know another obvious way). 
Don't know how correctly to patch the class because when I do that (with subclass) all calls to `super` goes crazy. Ideas?

Reply:
Ah.  Sorry; I should have paid more attention to the test body.

Probably the most straightforward way to do it is to write a test that calls `GetInputsCommand` with `security_level=1` (or `3`) and verify that the resulting addresses are correct.

Admittedly, the scope would be a too broad – since it also covers `AddressGenerator` functionality – but it's fairly straightforward to write, and it should make the test easier to understand.

Reply:
I learned a lot trying to do that! Yes, this is straightforward but tests would be slower. Another (non elegant) way would be to put some empty hook into __init__ of AddressGenerator and then patch it. Or introduce some global function like get_address_generator_class of get_address_generator (lots of refactoring). That's for tomorrow, now I go to sleep :)

Reply:
True, but that could cause more headaches down the road; give the [TDD series on Enterprise Craftsmanship](http://enterprisecraftsmanship.com/2015/06/29/test-induced-design-damage-or-why-tdd-is-so-painful/) a read for more info.

I'm OK with slow unit tests — the crypto tests already take up the bulk of it (especially if the optional C extension is not installed), so another few milliseconds to generate an address or two is just a drop in the ocean.

Reply:
so I created a test for all 3 level's of security, explicitly checking 3 addresses with static seed. It doesn't take very long. 
Thanks for the link above, it does make sense!